lib/spack/spack/package_base.py

```diff
@@ -1253,7 +1253,7 @@ def extendee_spec(self):
 
         # if the spec is concrete already, then it extends something
         # that is an *optional* dependency, and the dep isn't there.
-        if self.spec._concrete:
+        if self.spec._concrete and not self.spec.external:
             return None
```

Comment:
I created two spack instances, installed Python in the first, and then referenced it in the second like:

```
  packages:
    python:
      externals:
      - spec: python@3.10.8
        prefix: /path/to/other/spack/opt/spack/darwin-bigsur-skylake/apple-clang-12.0.0/python-3.10.8-m7zmavbe4v55ygbwl23frjr6nw5whi7e
      buildable: false
```

And was able to install py-cython, and furthermore in a `spack-python` shell `cython_spec.package.extendee_spec` returns the external Python instance (and `cython_spec['python'].concrete == True`).

Do you have externals other than Python? If so, can you mention them here?